torch/testing/_internal/common_device_type.py

```diff
@@ -275,6 +277,14 @@ def setUpClass(cls):
 if torch.cuda.is_available():
     device_type_test_bases.append(CUDATestBase)
 
+
+_PYTORCH_TEST_PATH = os.environ.get('PYTORCH_TEST_PATH', None)
+if _PYTORCH_TEST_PATH:
+    for path in _PYTORCH_TEST_PATH.split(':'):
+        mod = runpy.run_path(os.path.join(path, 'pytorch_test_base.py'), init_globals=globals())
```

Comment:
Let's have the environment variable fully specify the path to the file?

Reply:
I did it that way so that one could append to that path, and allow different device types to be added for tests.